ath79: add support for Airtight C-65 (Mojo Networks C-65)

vov4ik-il opened the pull request:
Airtight C-65 (Mojo Networks C-65) Dual radio, dual concurrent 2x2:2 MIMO 802.11ac Wave 1 access point.

Specification:

SoC	: Qualcomm Atheros QCA9558-AT4A
RAM	: DDR2 128 MiB
Flash	: SPI-NOR 2*16 MiB
WLAN	: 2.4/5 GHz 2T2R
2.4 GHz: QCA9558 (SoC)
5 GHz : QCA9882
Ethernet	: 1x 10/100/1000 Mbps
PHY : AR8035-A
LEDs/keys	: 4x/1x
UART: conn8 on PCB
conn8 =
 ----------------
| 1   2   3   4 |
 3.3v GND TX  RX
Settings: 115200 bps, no flow control

Flashing Instructions (TFTP):
1.Download "openwrt-ath79-generic-airtight_c-65-initramfs-kernel.bin" and "openwrt-ath79-generic-airtight_c-65-squashfs-sysupgrade.bin" and verify chcksums.
2. Setup TFTP sevrer with adress 192.168.1.100/24 on local machine connected to Mojo Networks C-65 device.
3. Rename "openwrt-ath79-generic-airtight_c-65-initramfs-kernel.bin" it to "c65_atnboot.bin" and put it in TFTP server root directory.
4. Powercycle the unit (very short power interruption is required, less than 1 second or it might not work) and wait about 90 seconds for the ram image to load.
5. Open a web browser and navigate to 192.168.1.1 for the router Web Interface and login with a blank password.
6. In the top menu click "System" -> "Backup/Flash Firmware",
7. Save current firmware: Select "firmware" in "Save mtdblock contents" and click "Save mtdblock" - this is your only chance to save your current firmware to be able to go back to the original state!!! No official frimware download is provided anywhere. I would do it twice to make sure the file is not corrupt due to ethernet error!
8. Flash OpenWRT frimware: In "Flash new firmware image" browse and select "openwrt-ath79-generic-airtight_c-65-squashfs-sysupgrade.bin", untick "Keep settings" and click "Flash firmware...", verify checksum with the original and click "Proceed"
9. Wait about 150 seconds for the router to flash (if you interrupt the power - you can brick it, however it is 80% recoverable if you repeat from step 4)
You just gave a new life to your Mojo Networks C-65 Wireless Access Point. Enjoy and considre donating to the project.

Return to original firmware:
Just flash the file you had backed up in step 7 of flashing instructions.

Known limitations in this release:
Wireless 5G LED behavior is inverted (steady "on" means the radio is disabled) as the LED connected to
the gpiochip on ath10k chip (QCA9882).
Only one 16Mb flash chip is available (the unit has 2).

adrianschmutzler commented on target/linux/ath79/image/generic.mk:
```diff
@@ -75,6 +75,16 @@ define Device/seama
 
+define Device/airtight_c-65
+  ATH_SOC := qca9558
+  DEVICE_TITLE := Airtight C-65
```
Please update to use DEVICE_VENDOR, DEVICE_MODEL and DEVICE_VARIANT as introduced in:
https://github.com/openwrt/openwrt/commit/8c3dbaf8b49233e275b2dce4e30dc0a2bf8c86c5

vov4ik-il replied:
Done

adrianschmutzler commented:
I'm currently working on a PR to store the label-mac-address in the DTS (https://github.com/openwrt/openwrt/pull/2159).

I would appreciate if you could provide information about your device:
Please run
`find /proc/device-tree/ -name "*mac-address*"`
on your device (just with the current firmware, no need to update anything; just has to be ath79, no ar71xx).
For each of the returned paths (if there are any), retrieve the mac-address with `cat`, e.g.
`cat /proc/device-tree/ahb/eth@19000000/mac-address`

Now check which of those returned MAC addresses matches the one on the label/sticker/case of your device and tell me the device-tree path leading to it. Note that you do not have to post your MAC addresses here, but just a statement about which match, e.g. like
```
/proc/device-tree/ahb/eth@1a000000/mac-address
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address-increment
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address
/proc/device-tree/ahb/eth@19000000/mac-address <== this one
/proc/device-tree/ahb/eth@19000000/mtd-mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mtd-mac-address
```

There may be one, two or no paths giving the correct address. It may also happen that the `find` command returns nothing. Thanks.

adrianschmutzler commented:
Unfortunately, my description is slightly incorrect. The MAC addresses are stored binary, so you can't read with `cat`, but have to use the following code to read from the paths retrieved by `find`:
```
. /lib/functions/system.sh
get_mac_binary "$yourpath" 0
```
The include is obviously only required once. Sorry.

vov4ik-il commented:
Hi,
I’ll may get a better answer in a couple of days, but the factory mac is stored (binary) at the beginning of the ART partition on this specific device.

adrianschmutzler commented:
@vov4ik-il 
So, it would be both
```
/proc/device-tree/ahb/eth@19000000/mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mac-address
```
containing the label mac address then.

vov4ik-il commented:
Checked the label and parsed the rom, it is found at 0x00 and 0x20 as binary and so it is on the label as both S/N and MAC.

adrianschmutzler commented:
Thanks, so depending on which PR is faster, the slower one will have to include `label-mac-device = &eth0;` for this device.

ynezz commented on target/linux/ath79/base-files/etc/board.d/01_leds:
```diff
@@ -8,6 +8,11 @@ board=$(board_name)
 case "$board" in
+airtight,c-65)
+	ucidef_set_led_netdev "lan" "LAN" "$boardname:green:lan" "eth0"
+	ucidef_set_led_wlan "wlan2g" "WiFi 2.4GHz" "$boardname:green:wlan2g" "phy1tpt"
```
Drop the wlan2g/wlan5g lines and use `linux,default-trigger = "phy0tpt";` and `linux,default-trigger = "phy1tpt";` in your LED nodes instead. Configuration of LEDs in DTS is preferred. Example:
https://github.com/openwrt/openwrt/blob/e65f935de9e67ea07a251f114455f12e9555c5bf/target/linux/ath79/dts/ar9344_comfast_cf-e120a-v3.dts#L36-L40

tmn505 replied:
If there is no GPIO from SoC controlling the wlan5g led then it should stay defined in this file.

vov4ik-il replied:
Indeed no GPIO on the SoC, it is on the WLAN card.

ynezz commented on target/linux/ath79/dts/qca9558_airtight_c-65.dts:
```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+/*
+ * This board has two identical flash ICs but only one is defined in this file.
```
Remove this comment and place it into your commit message.

vov4ik-il replied:
Done.

ynezz commented on target/linux/ath79/dts/qca9558_airtight_c-65.dts:
```diff
@@ -0,0 +1,150 @@
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
```
Please try this with `compatible = "gpio-keys";` which is interrupt based variant, and remove that `poll-interval` property

vov4ik-il replied:
Done.

ynezz commented:
Please don't close this PR and open a new one. Please use `git commit --amend` to add changes to your commit. Use `git push --force` to overwrite your github repository/branch. The PR will afterwards update automatically. See the [squashing commits instructions](https://openwrt.org/docs/guide-developer/working-with-github-pr#squashing_commits) on how to modify this PR without closing it and open a new one.

You need to squash all 4 commits into 1 commit `ath79: add support for Airtight C-65 (Mojo Networks C-65)`, take care about max. line length of 75 chars in the commit description as well, please check [Submitting patches](https://openwrt.org/submitting-patches#submitting_patches) for details.

ynezz commented on target/linux/ath79/dts/qca9558_airtight_c-65.dts:
```diff
@@ -32,7 +32,7 @@
 			label = "c-65:amber:power";
 			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
-			default-state = "on";
+			default-state = "off";
```
Remove this property, 'off' is the default state, no need to set it explicitly again.

tmn505 replied:
As this is power LED by default it should be 'on', and left to user to configure it to desired state in `/etc/config/system`.

vov4ik-il replied:
It is "off" to match the others, the user can change everything to his taste afterward.

vov4ik-il replied:
Property wiped.

tmn505 commented on target/linux/ath79/dts/qca9558_airtight_c-65.dts:
```diff
@@ -0,0 +1,150 @@
+
+&pcie1 {
+	status = "okay";
+};
```
You have two pcie node enabled but probably only one is used/populated on the board. Please remove unused one. Also add 'wifi' subnode, looking like this:
````
wifi@0,0 {
                compatible = "pci168c,003c";
                reg = <0x0000 0 0 0 0>;
        };
````
with proper VID and PID.

vov4ik-il replied:
Removed unused, did not add the section - the board is not with me right now.

tmn505 commented on target/linux/ath79/dts/qca9558_airtight_c-65.dts:
```diff
@@ -0,0 +1,150 @@
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "macronix, mx25l12805d", "jedec,spi-nor";
```
Remove "macronix, mx25l12805d" compatible, as other production batches could have different chips.

vov4ik-il replied:
Removed

tmn505 commented on target/linux/ath79/dts/qca9558_airtight_c-65.dts:
```diff
@@ -0,0 +1,150 @@
+	status = "okay";
+	mtd-cal-data = <&art 0x1000>;
+	mtd-mac-address = <&art 0x0>;
+};
```
EOF should be on new line.

vov4ik-il replied:
done

tmn505 commented on target/linux/ath79/image/generic.mk:
```diff
@@ -75,6 +75,17 @@ define Device/seama
+  DEVICE_VENDOR := Airtight
+  DEVICE_MODEL := C-65
+  DEVICE_PACKAGES := kmod-ath10k ath10k-firmware-qca988x
+  IMAGE/default := append-kernel | pad-to $$$$(BLOCKSIZE) | append-rootfs | pad-rootfs | check-size $$$$(IMAGE_SIZE)
```
Why is this defined when not used/called anywhere?

vov4ik-il replied:
Removed

tmn505 commented on target/linux/ath79/image/generic.mk:
```diff
@@ -75,6 +75,17 @@ define Device/seama
   SEAMA_SIGNATURE :=
 endef
 
+define Device/airtight_c-65
```
Device recipes should be sorted by alphabetic order.

vov4ik-il replied:
I did not put Seama on top. The rest is alphabetic.

tmn505 replied:
The seama is common define used by few devices so yes, it should be on above them, but definition for Your device is above `8dev_carambola2` and digits should be before letters.

Then fix commit title, SoB line and rebase on top of master and that should be it.
Good luck.

tmn505 commented on target/linux/ath79/image/generic.mk:
```diff
@@ -75,6 +75,17 @@ define Device/seama
+  DEVICE_PACKAGES := kmod-ath10k ath10k-firmware-qca988x
+  IMAGE/default := append-kernel | pad-to $$$$(BLOCKSIZE) | append-rootfs | pad-rootfs | check-size $$$$(IMAGE_SIZE)
+  IMAGE_SIZE := 15936k
+  IMAGE/sysupgrade.bin := append-kernel | append-rootfs | pad-rootfs | append-metadata | check-size $$$$(IMAGE_SIZE)
```
This is redefinition of image recipe in https://github.com/openwrt/openwrt/blob/d616b2c906690d2e471144ca12b0a9ed28de21c2/target/linux/ath79/image/Makefile#L65-L66 which differs only in padding kernel to block size (which generally should be done). Is there a reason for this?

vov4ik-il replied:
Removed

Commit:
Signed-off-by: Vladimir Georgievsky vladimir.georgievsky@yahoo.com

Airtight C-65 (Mojo Networks C-65) Dual radio, dual concurrent 2x2:2 MIMO 802.11ac Wave 1 access point based on Qualcomm
Atheros QCA9558.

Specification:

SoC	: Qualcomm Atheros QCA9558-AT4A
RAM	: DDR2 128 MiB
Flash	: SPI-NOR 2*16 MiB
WLAN	: 2.4/5 GHz 2T2R
2.4 GHz: QCA9558 (SoC)
5 GHz : QCA9882
Ethernet	: 1x 10/100/1000 Mbps
PHY : AR8035-A
LEDs/keys	: 6x/1x (4 LEDs, 2 dual color)
UART: conn8 on PCB
conn8 =
 ----------------
| 1   2   3   4 |
 3.3v GND TX  RX
Settings: 115200 bps, no flow control

Flashing Instructions (TFTP):
1. Download "openwrt-ath79-generic-airtight_c-65-initramfs-kernel.bin"
 and "openwrt-ath79-generic-airtight_c-65-squashfs-sysupgrade.bin" and
 verify chcksums.
2. Setup TFTP sevrer with adress 192.168.1.100/24 on local machine
 connected to Mojo Networks C-65 device.
3. Rename "openwrt-ath79-generic-airtight_c-65-initramfs-kernel.bin" it
 to "c65_atnboot.bin" and put it in TFTP server root directory.
4. Powercycle the unit (very short power interruption is required, less
 than 1 second or it might not work) and wait about 90 seconds for the
 ram image to load.
5. Open a web browser and navigate to 192.168.1.1 for the router Web
 Interface and login with a blank password.
6. In the top menu click "System" -> "Backup/Flash Firmware",
7. Save current firmware: Select "firmware" in "Save mtdblock contents"
 and click "Save mtdblock" - this is your only chance to save your
 current firmware to be able to go back to the original state!!!
 No official frimware download is provided anywhere. I would do it twice
 to make sure the file is not corrupt due to ethernet error!
8. Flash OpenWRT frimware: In "Flash new firmware image" browse and
 select "openwrt-ath79-generic-airtight_c-65-squashfs-sysupgrade.bin",
 untick "Keep settings" and click "Flash firmware...", verify checksum
 with the original and click "Proceed"
9. Wait about 150 seconds for the router to flash (if you interrupt the
 power - you can brick it, however it is 80% recoverable if you repeat
 from step 4)

You just gave a new life to your Mojo Networks C-65 Wireless Access Point.
Enjoy and considre donating to the project.

Back to original firmware:
Flash the file you had backed up in step 7 of flashing instructions.

Known limitations in this release:
Wireless 5G LED behavior is inverted
 (steady "on" means the radio is disabled), the LED is on QCA9882.
 As a workaround the configuration is tailored so that all LEDs match
 this behavior.
Only one 16Mb flash chip is available (the unit has 2).

adrianschmutzler commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -9,6 +9,9 @@ ath79_setup_interfaces()
 	case "$board" in
+	airtight,c-65)
+		ucidef_set_interface_lan "eth0"
+		;;
```
Looks like this case can be merged with the subsequent one.

adrianschmutzler commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -262,6 +265,9 @@ ath79_setup_macs()
 
 	case "$board" in
+	airtight,c65)
+		lan_mac=$(macaddr_setbit_la "$(mtd_get_mac_binary art 0)")
```
Why are you setting the local bit for lan here (&eth0 is set to art 0x0 already)?

adrianschmutzler replied:
To me it looks like that line can be dropped.

adrianschmutzler commented on target/linux/ath79/dts/qca9558_airtight_c-65.dts:
```diff
@@ -0,0 +1,140 @@
+	status = "okay";
+	mtd-cal-data = <&art 0x1000>;
+	mtd-mac-address = <&art 0x0>;
+
```
Remove empty line.

adrianschmutzler commented:
@vov4ik-il Do you plan to proceed with this PR?

vov4ik-il commented:
> @vov4ik-il Do you plan to proceed with this PR?

Adrian, I removed the linux VM I was working on and got pretty busy with my main work. Since there is not much interest in this in the community - I guess I will leave it as it is. You are more than welcome to finish if you would like to, or I can just close it. I still do have a few devices if things need to be tested.

adrianschmutzler commented:
Okay, then I will close this.

Feel free to ask for reopening if you want to proceed (when you have more time).

deppen12 commented:
i have some problem with my c-65. its stop working after 10-15 min. orange light start blinking and every thing stop working , lan not works nor wifi

vov4ik-il commented:
Hi, please attach the logs and your build .bin, I might still have a device I can test stuff on… but no promises made :-)Best regards,This message was sent from iPhone.On Oct 13, 2023, at 6:27 AM, deppen12 ***@***.***> wrote:﻿
i have some problem with my c-65. its stop working after 10-15 min. orange light start blinking and every thing stop working , lan not works nor wifi

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

deppen12 commented:
now its stop working . firmware is gone. i downloaded firmware from this link "https://github.com/vov4ik-il/Mojo-Networks-C-65"

log
=================================================

U-Boot 1.1.4--LSDK-10.2.85 (Sep  3 2014 - 16:45:27)

EAP1018A1-FLF-FW - Scorpion 1.0

DRAM:  128 MB

cpu_pll_cfg: 0x21480  cpu_pll_dither: 0x3c103f
CFG_PLL_FREQ: 0x2  CFG_HZ: 360000000
CPU Clock: 720MHz    DDR Clock: 680MHz

Flash Manuf Id 0xc2, DeviceId0 0x20, DeviceId1 0x18
flash size 16MB, sector count = 256
flash-2 size 16MB, sector count = 256
Flash: 16 *2 MB
In:    serial
Out:   serial
Err:   serial
Net:   ath_gmac_enet_initialize...
athrs_sgmii_res_cal: cal value = 0xe
Fetching MAC Address from 0x87fedb6c
ath_gmac_enet_initialize: reset mask:c02200
Scorpion  ----> 8035 PHY *
: cfg1 0x80000000 cfg2 0x7114
eth0: 00:11:74:e7:dd:1f
ATHR_AUTONEG_ADVERT:1DE1
ATHR_1000BASET_CONTROL:200
ATHR_PHY_CONTROL:3100
ATHRSF1_PHY: Port 5, Neg Success
ATHRSF1_PHY: unit 0 phy addr 5 eth0 up
eth0
Setting 0x18116290 to 0x810214f
Hit any key to stop autoboot:  0
multiboot=1
Trying eth0
eth0 link down
FAIL
Using eth0 device
TFTP from server 192.168.1.100; our IP address is 192.168.1.1
Filename 'c65_atnboot.bin'.
Load address: 0x81000000
Loading: Tx Timed out
T Tx Timed out

Retry count exceeded; starting again
Booting from internal Flash...
## Booting image at 9f060000 ...
   Image Name:   MIPS OpenWrt Linux-4.14.121
   Created:      2019-05-31  19:01:33 UTC
   Image Type:   MIPS Linux Kernel Image (lzma compressed)
   Data Size:    1618859 Bytes =  1.5 MB
   Load Address: 80060000
   Entry Point:  80060000
   Verifying Checksum at 0x9f060040 ...OK
   Uncompressing Kernel Image ... OK
No initrd
## Transferring control to Linux (at address 80060000) ...
## Giving linux memsize in bytes, 134217728

Starting kernel ...

[    0.000000] Linux version 4.14.121 (osboxes@osboxes) (gcc version 7.4.0 (Open                                                                                                                                                             Wrt GCC 7.4.0 r10114-a1210f8888)) #0 Fri May 31 19:01:33 2019
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019750 (MIPS 74Kc)
[    0.000000] MIPS: machine is AirTight C-65
[    0.000000] SoC: Qualcomm Atheros QCA9558 ver 1 rev 0
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 08000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32                                                                                                                                                              bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] random: get_random_bytes called from start_kernel+0x90/0x478 with                                                                                                                                                              crng_init=0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32512
[    0.000000] Kernel command line: console=ttyS0,115200 rootfstype=squashfs,jff                                                                                                                                                             s2
[    0.000000] PID hash table entries: 512 (order: -1, 2048 bytes)
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 123184K/131072K available (3792K kernel code, 151K rwdata                                                                                                                                                             , 912K rodata, 1216K init, 204K bss, 7888K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 720.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_i                                                                                                                                                             dle_ns: 5309056796 ns
[    0.000007] sched_clock: 32 bits at 360MHz, resolution 2ns, wraps every 59652                                                                                                                                                             32126ns
[    0.008288] Calibrating delay loop... 358.80 BogoMIPS (lpj=1794048)
[    0.074976] pid_max: default: 32768 minimum: 301
[    0.080050] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.087086] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.097819] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, ma                                                                                                                                                             x_idle_ns: 19112604462750000 ns
[    0.108312] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.114853] pinctrl core: initialized pinctrl subsystem
[    0.121116] NET: Registered protocol family 16
[    0.131376] PCI host bridge /ahb/apb/pcie-controller@180c0000 ranges:
[    0.138242]  MEM 0x0000000010000000..0x0000000011ffffff
[    0.143824]   IO 0x0000000000000000..0x0000000000000000
[    0.149677] PCI host bridge /ahb/apb/pcie-controller@18250000 ranges:
[    0.156561]  MEM 0x0000000012000000..0x0000000013ffffff
[    0.162114]   IO 0x0000000000000001..0x0000000000000001
[    0.167675] ar724x-pci 18250000.pcie-controller: PCIe link is down
[    0.186719] PCI host bridge to bus 0000:00
[    0.191117] pci_bus 0000:00: root bus resource [mem 0x10000000-0x11ffffff]
[    0.198426] pci_bus 0000:00: root bus resource [io  0x0000]
[    0.204356] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.211572] pci_bus 0000:00: No busn resource found for root bus, will use [b                                                                                                                                                             us 00-ff]
[    0.220413] pci 0000:00:00.0: BAR 0: assigned [mem 0x10000000-0x101fffff 64bi                                                                                                                                                             t]
[    0.228206] pci 0000:00:00.0: BAR 6: assigned [mem 0x10200000-0x1020ffff pref                                                                                                                                                             ]
[    0.236030] PCI host bridge to bus 0000:01
[    0.240387] pci_bus 0000:01: root bus resource [mem 0x12000000-0x13ffffff]
[    0.247712] pci_bus 0000:01: root bus resource [io  0x0001]
[    0.253631] pci_bus 0000:01: root bus resource [??? 0x00000000 flags 0x0]
[    0.260855] pci_bus 0000:01: No busn resource found for root bus, will use [b                                                                                                                                                             us 01-ff]
[    0.271276] clocksource: Switched to clocksource MIPS
[    0.277433] NET: Registered protocol family 2
[    0.282685] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.290108] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.296909] TCP: Hash tables configured (established 1024 bind 1024)
[    0.303771] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.309985] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.316876] NET: Registered protocol family 1
[    0.323680] Crashlog allocated RAM at address 0x3f00000
[    0.330125] workingset: timestamp_bits=30 max_order=15 bucket_order=0
[    0.340827] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.347061] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORIT                                                                                                                                                             Y) (c) 2001-2006 Red Hat, Inc.
[    0.363761] io scheduler noop registered
[    0.367935] io scheduler deadline registered (default)
[    0.375159] pinctrl-single 1804002c.pinmux: 544 pins at pa b804002c size 68
[    0.383436] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.390738] console [ttyS0] disabled
[    0.394603] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 250                                                                                                                                                             0000) is a 16550A
[    0.403792] console [ttyS0] enabled
[    0.403792] console [ttyS0] enabled
[    0.411195] bootconsole [early0] disabled
[    0.411195] bootconsole [early0] disabled
[    0.424241] m25p80 spi0.0: mx25l12805d (16384 Kbytes)
[    0.429401] 4 fixed-partitions partitions found on MTD device spi0.0
[    0.435872] Creating 4 MTD partitions on "spi0.0":
[    0.440737] 0x000000000000-0x000000040000 : "u-boot"
[    0.446423] 0x000000040000-0x000000060000 : "uboot-env"
[    0.452361] 0x000000060000-0x000000ff0000 : "firmware"
[    0.460111] 2 uimage-fw partitions found on MTD device firmware
[    0.466155] Creating 2 MTD partitions on "firmware":
[    0.471193] 0x000000000000-0x00000018b3eb : "kernel"
[    0.476830] 0x00000018b3eb-0x000000f90000 : "rootfs"
[    0.482477] mtd: device 4 (rootfs) set to be root filesystem
[    0.488246] 1 squashfs-split partitions found on MTD device rootfs
[    0.494549] 0x000000490000-0x000000f90000 : "rootfs_data"
[    0.500604] 0x000000ff0000-0x000001000000 : "art"
[    0.506856] libphy: Fixed MDIO Bus: probed
[    0.512910] ag71xx 19000000.eth: invalid MAC address, using random address
[    0.852967] libphy: ag71xx_mdio: probed
[    0.952008] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:05 [uid=004dd                                                                                                                                                             072, driver=Atheros 8035 ethernet]
[    0.962754] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: rgmii-id
[    0.970897] NET: Registered protocol family 10
[    0.978397] Segment Routing with IPv6
[    0.982229] NET: Registered protocol family 17
[    0.986776] 8021q: 802.1Q VLAN Support v1.8
[    0.997828] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
[    1.008642] Freeing unused kernel memory: 1216K
[    1.013263] This architecture does not have kernel memory protection.
[    1.689308] init: Console is alive
[    1.692967] init: - watchdog -
[    2.401288] random: fast init done
[    2.920673] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    2.949615] kmodloader: done loading kernel modules from /etc/modules-boot.d/                                                                                                                                                             *
[    2.958204] init: - preinit -
[    4.108926] random: jshn: uninitialized urandom read (4 bytes read)
[    4.219282] random: jshn: uninitialized urandom read (4 bytes read)
[    4.256282] random: jshn: uninitialized urandom read (4 bytes read)
[    4.531784] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    7.764029] mount_root: loading kmods from internal overlay
[    7.801419] kmodloader: loading kernel modules from //etc/modules-boot.d/*
[    7.808678] kmodloader: done loading kernel modules from //etc/modules-boot.d                                                                                                                                                             /*
[    8.269491] jffs2: notice: (413) jffs2_build_xattr_subsystem: complete buildi                                                                                                                                                             ng xattr subsystem, 51 of xdatum (3 unchecked, 48 orphan) and 215 of xref (40 de                                                                                                                                                             ad, 120 orphan) found.
[    8.286163] block: attempting to load /tmp/jffs_cfg/upper/etc/config/fstab
[    8.299453] block: extroot: not configured
[    8.456932] jffs2: notice: (411) jffs2_build_xattr_subsystem: complete buildi                                                                                                                                                             ng xattr subsystem, 51 of xdatum (3 unchecked, 48 orphan) and 215 of xref (40 de                                                                                                                                                             ad, 120 orphan) found.
[    8.476157] mount_root: loading kmods from internal overlay
[    8.509413] kmodloader: loading kernel modules from /tmp/overlay/upper/etc/mo                                                                                                                                                             dules-boot.d/*
[    8.519767] kmodloader: done loading kernel modules from /tmp/overlay/upper/e                                                                                                                                                             tc/modules-boot.d/*
[    8.734605] block: attempting to load /tmp/jffs_cfg/upper/etc/config/fstab
[    8.745691] block: extroot: not configured
[    8.750619] mount_root: switching to jffs2 overlay
[    8.783412] overlayfs: upper fs does not support tmpfile.
[    8.793452] urandom-seed: Seeding with /etc/urandom.seed
Error relocating /bin/cat: __gmtime64_r: symbol not found
Error relocating /bin/cat: __mktime64: symbol not found
Error relocating /bin/cat: __fstat_time64: symbol not found
Error relocating /bin/cat: __clock_settime64: symbol not found
Error relocating /bin/cat: __clock_gettime64: symbol not found
Error relocating /bin/cat: __utimes_time64: symbol not found
Error relocating /bin/cat: __futimens_time64: symbol not found
Error relocating /bin/cat: __utimensat_time64: symbol not found
Error relocating /bin/cat: __localtime64_r: symbol not found
Error relocating /bin/cat: __wait3_time64: symbol not found
Error relocating /bin/cat: __stat_time64: symbol not found
Error relocating /bin/cat: __time64: symbol not found
Error relocating /bin/cat: __nanosleep_time64: symbol not found
Error relocating /bin/cat: __gettimeofday_time64: symbol not found
Error relocating /bin/cat: __[    9.027665] procd: - early -
settimeofday_tim[    9.032679] procd: - watchdog -
e64: symbol not found
Error relocating /bin/cat: __localtime64: symbol not found
Error relocating /bin/cat: __adjtimex_time64: symbol not found
Error relocating /bin/cat: __lstat_time64: symbol not found
Error relocating /bin/cat: __ctime64: symbol not found
Error relocating /bin/cat: __setitimer_time64: symbol not found
Error relocating /bin/grep: __gmtime64_r: symbol not found
Error relocating /bin/grep: __mktime64: symbol not found
Error relocating /bin/grep: __fstat_time64: symbol not found
Error relocating /bin/grep: __clock_settime64: symbol not found
Error relocating /bin/grep: __clock_gettime64: symbol not found
Error relocating /bin/grep: __utimes_time64: symbol not found
Error relocating /bin/grep: __futimens_time64: symbol not found
Error relocating /bin/grep: __utimensat_time64: symbol not found
Error relocating /bin/grep: __localtime64_r: symbol not found
Error relocating /bin/grep: __wait3_time64: symbol not found
Error relocating /bin/grep: __stat_time64: symbol not found
Error relocating /bin/grep: __time64: symbol not found
Error relocating /bin/grep: __nanosleep_time64: symbol not found
Error relocating /bin/grep: __gettimeofday_time64: symbol not found
Error relocating /bin/grep: __settimeofday_time64: symbol not found
Error relocating /bin/grep: __localtime64: symbol not found
Error relocating /bin/grep: __adjtimex_time64: symbol not found
Error relocating /bin/grep: __lstat_time64: symbol not found
Error relocating /bin/grep: __ctime64: symbol not found
Error relocating /bin/grep: __setitimer_time64: symbol not found
[    9.628477] procd: - watchdog -
[    9.631939] procd: - ubus -
[    9.785657] urandom_read: 4 callbacks suppressed
[    9.785663] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.799462] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.806606] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.813851] procd: - init -
Please press Enter to activate this console.

Error relocating /usr/libexec/loPlease press Enter to activate this console.
==============================================================

vov4ik-il commented:
It did not flash or loaded garbage from the other flash chip.Either way, follow the readme and repeat the installation twice. This message was sent from iPhone.On Oct 15, 2023, at 4:35 AM, deppen12 ***@***.***> wrote:﻿
now its stop working . firmware is gone. i downloaded firmware from this link "https://github.com/vov4ik-il/Mojo-Networks-C-65"
log
U-Boot 1.1.4--LSDK-10.2.85 (Sep  3 2014 - 16:45:27)
EAP1018A1-FLF-FW - Scorpion 1.0
DRAM:  128 MB
cpu_pll_cfg: 0x21480  cpu_pll_dither: 0x3c103f
CFG_PLL_FREQ: 0x2  CFG_HZ: 360000000
CPU Clock: 720MHz    DDR Clock: 680MHz
Flash Manuf Id 0xc2, DeviceId0 0x20, DeviceId1 0x18
flash size 16MB, sector count = 256
flash-2 size 16MB, sector count = 256
Flash: 16 *2 MB
In:    serial
Out:   serial
Err:   serial
Net:   ath_gmac_enet_initialize...
athrs_sgmii_res_cal: cal value = 0xe
Fetching MAC Address from 0x87fedb6c
ath_gmac_enet_initialize: reset mask:c02200
Scorpion  ----> 8035 PHY *
: cfg1 0x80000000 cfg2 0x7114
eth0: 00:11:74:e7:dd:1f
ATHR_AUTONEG_ADVERT:1DE1
ATHR_1000BASET_CONTROL:200
ATHR_PHY_CONTROL:3100
ATHRSF1_PHY: Port 5, Neg Success
ATHRSF1_PHY: unit 0 phy addr 5 eth0 up
eth0
Setting 0x18116290 to 0x810214f
Hit any key to stop autoboot:  0
multiboot=1
Trying eth0
eth0 link down
FAIL
Using eth0 device
TFTP from server 192.168.1.100; our IP address is 192.168.1.1
Filename 'c65_atnboot.bin'.
Load address: 0x81000000
Loading: Tx Timed out
T Tx Timed out
Retry count exceeded; starting again
Booting from internal Flash...
Booting image at 9f060000 ...
Image Name:   MIPS OpenWrt Linux-4.14.121
Created:      2019-05-31  19:01:33 UTC
Image Type:   MIPS Linux Kernel Image (lzma compressed)
Data Size:    1618859 Bytes =  1.5 MB
Load Address: 80060000
Entry Point:  80060000
Verifying Checksum at 0x9f060040 ...OK
Uncompressing Kernel Image ... OK
No initrd
Transferring control to Linux (at address 80060000) ...
Giving linux memsize in bytes, 134217728
Starting kernel ...
[    0.000000] Linux version 4.14.121 ***@***.***) (gcc version 7.4.0 (Open                                                                                                                                                             Wrt GCC 7.4.0 r10114-a1210f8888)) #0 Fri May 31 19:01:33 2019
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019750 (MIPS 74Kc)
[    0.000000] MIPS: machine is AirTight C-65
[    0.000000] SoC: Qualcomm Atheros QCA9558 ver 1 rev 0
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 08000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32                                                                                                                                                              bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] random: get_random_bytes called from start_kernel+0x90/0x478 with                                                                                                                                                              crng_init=0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32512
[    0.000000] Kernel command line: console=ttyS0,115200 rootfstype=squashfs,jff                                                                                                                                                             s2
[    0.000000] PID hash table entries: 512 (order: -1, 2048 bytes)
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 123184K/131072K available (3792K kernel code, 151K rwdata                                                                                                                                                             , 912K rodata, 1216K init, 204K bss, 7888K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 720.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_i                                                                                                                                                             dle_ns: 5309056796 ns
[    0.000007] sched_clock: 32 bits at 360MHz, resolution 2ns, wraps every 59652                                                                                                                                                             32126ns
[    0.008288] Calibrating delay loop... 358.80 BogoMIPS (lpj=1794048)
[    0.074976] pid_max: default: 32768 minimum: 301
[    0.080050] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.087086] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.097819] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, ma                                                                                                                                                             x_idle_ns: 19112604462750000 ns
[    0.108312] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.114853] pinctrl core: initialized pinctrl subsystem
[    0.121116] NET: Registered protocol family 16
[    0.131376] PCI host bridge ***@***.*** ranges:
[    0.138242]  MEM 0x0000000010000000..0x0000000011ffffff
[    0.143824]   IO 0x0000000000000000..0x0000000000000000
[    0.149677] PCI host bridge ***@***.*** ranges:
[    0.156561]  MEM 0x0000000012000000..0x0000000013ffffff
[    0.162114]   IO 0x0000000000000001..0x0000000000000001
[    0.167675] ar724x-pci 18250000.pcie-controller: PCIe link is down
[    0.186719] PCI host bridge to bus 0000:00
[    0.191117] pci_bus 0000:00: root bus resource [mem 0x10000000-0x11ffffff]
[    0.198426] pci_bus 0000:00: root bus resource [io  0x0000]
[    0.204356] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.211572] pci_bus 0000:00: No busn resource found for root bus, will use [b                                                                                                                                                             us 00-ff]
[    0.220413] pci 0000:00:00.0: BAR 0: assigned [mem 0x10000000-0x101fffff 64bi                                                                                                                                                             t]
[    0.228206] pci 0000:00:00.0: BAR 6: assigned [mem 0x10200000-0x1020ffff pref                                                                                                                                                             ]
[    0.236030] PCI host bridge to bus 0000:01
[    0.240387] pci_bus 0000:01: root bus resource [mem 0x12000000-0x13ffffff]
[    0.247712] pci_bus 0000:01: root bus resource [io  0x0001]
[    0.253631] pci_bus 0000:01: root bus resource [??? 0x00000000 flags 0x0]
[    0.260855] pci_bus 0000:01: No busn resource found for root bus, will use [b                                                                                                                                                             us 01-ff]
[    0.271276] clocksource: Switched to clocksource MIPS
[    0.277433] NET: Registered protocol family 2
[    0.282685] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.290108] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.296909] TCP: Hash tables configured (established 1024 bind 1024)
[    0.303771] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.309985] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.316876] NET: Registered protocol family 1
[    0.323680] Crashlog allocated RAM at address 0x3f00000
[    0.330125] workingset: timestamp_bits=30 max_order=15 bucket_order=0
[    0.340827] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.347061] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORIT                                                                                                                                                             Y) (c) 2001-2006 Red Hat, Inc.
[    0.363761] io scheduler noop registered
[    0.367935] io scheduler deadline registered (default)
[    0.375159] pinctrl-single 1804002c.pinmux: 544 pins at pa b804002c size 68
[    0.383436] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.390738] console [ttyS0] disabled
[    0.394603] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 250                                                                                                                                                             0000) is a 16550A
[    0.403792] console [ttyS0] enabled
[    0.403792] console [ttyS0] enabled
[    0.411195] bootconsole [early0] disabled
[    0.411195] bootconsole [early0] disabled
[    0.424241] m25p80 spi0.0: mx25l12805d (16384 Kbytes)
[    0.429401] 4 fixed-partitions partitions found on MTD device spi0.0
[    0.435872] Creating 4 MTD partitions on "spi0.0":
[    0.440737] 0x000000000000-0x000000040000 : "u-boot"
[    0.446423] 0x000000040000-0x000000060000 : "uboot-env"
[    0.452361] 0x000000060000-0x000000ff0000 : "firmware"
[    0.460111] 2 uimage-fw partitions found on MTD device firmware
[    0.466155] Creating 2 MTD partitions on "firmware":
[    0.471193] 0x000000000000-0x00000018b3eb : "kernel"
[    0.476830] 0x00000018b3eb-0x000000f90000 : "rootfs"
[    0.482477] mtd: device 4 (rootfs) set to be root filesystem
[    0.488246] 1 squashfs-split partitions found on MTD device rootfs
[    0.494549] 0x000000490000-0x000000f90000 : "rootfs_data"
[    0.500604] 0x000000ff0000-0x000001000000 : "art"
[    0.506856] libphy: Fixed MDIO Bus: probed
[    0.512910] ag71xx 19000000.eth: invalid MAC address, using random address
[    0.852967] libphy: ag71xx_mdio: probed
[    0.952008] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:05 [uid=004dd                                                                                                                                                             072, driver=Atheros 8035 ethernet]
[    0.962754] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: rgmii-id
[    0.970897] NET: Registered protocol family 10
[    0.978397] Segment Routing with IPv6
[    0.982229] NET: Registered protocol family 17
[    0.986776] 8021q: 802.1Q VLAN Support v1.8
[    0.997828] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
[    1.008642] Freeing unused kernel memory: 1216K
[    1.013263] This architecture does not have kernel memory protection.
[    1.689308] init: Console is alive
[    1.692967] init: - watchdog -
[    2.401288] random: fast init done
[    2.920673] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    2.949615] kmodloader: done loading kernel modules from /etc/modules-boot.d/                                                                                                                                                             *
[    2.958204] init: - preinit -
[    4.108926] random: jshn: uninitialized urandom read (4 bytes read)
[    4.219282] random: jshn: uninitialized urandom read (4 bytes read)
[    4.256282] random: jshn: uninitialized urandom read (4 bytes read)
[    4.531784] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Press the [f] key and hit [enter] to enter failsafe mode
Press the [1], [2], [3] or [4] key and hit [enter] to select the debug level
[    7.764029] mount_root: loading kmods from internal overlay
[    7.801419] kmodloader: loading kernel modules from //etc/modules-boot.d/*
[    7.808678] kmodloader: done loading kernel modules from //etc/modules-boot.d                                                                                                                                                             /*
[    8.269491] jffs2: notice: (413) jffs2_build_xattr_subsystem: complete buildi                                                                                                                                                             ng xattr subsystem, 51 of xdatum (3 unchecked, 48 orphan) and 215 of xref (40 de                                                                                                                                                             ad, 120 orphan) found.
[    8.286163] block: attempting to load /tmp/jffs_cfg/upper/etc/config/fstab
[    8.299453] block: extroot: not configured
[    8.456932] jffs2: notice: (411) jffs2_build_xattr_subsystem: complete buildi                                                                                                                                                             ng xattr subsystem, 51 of xdatum (3 unchecked, 48 orphan) and 215 of xref (40 de                                                                                                                                                             ad, 120 orphan) found.
[    8.476157] mount_root: loading kmods from internal overlay
[    8.509413] kmodloader: loading kernel modules from /tmp/overlay/upper/etc/mo                                                                                                                                                             dules-boot.d/*
[    8.519767] kmodloader: done loading kernel modules from /tmp/overlay/upper/e                                                                                                                                                             tc/modules-boot.d/*
[    8.734605] block: attempting to load /tmp/jffs_cfg/upper/etc/config/fstab
[    8.745691] block: extroot: not configured
[    8.750619] mount_root: switching to jffs2 overlay
[    8.783412] overlayfs: upper fs does not support tmpfile.
[    8.793452] urandom-seed: Seeding with /etc/urandom.seed
Error relocating /bin/cat: __gmtime64_r: symbol not found
Error relocating /bin/cat: __mktime64: symbol not found
Error relocating /bin/cat: __fstat_time64: symbol not found
Error relocating /bin/cat: __clock_settime64: symbol not found
Error relocating /bin/cat: __clock_gettime64: symbol not found
Error relocating /bin/cat: __utimes_time64: symbol not found
Error relocating /bin/cat: __futimens_time64: symbol not found
Error relocating /bin/cat: __utimensat_time64: symbol not found
Error relocating /bin/cat: __localtime64_r: symbol not found
Error relocating /bin/cat: __wait3_time64: symbol not found
Error relocating /bin/cat: __stat_time64: symbol not found
Error relocating /bin/cat: __time64: symbol not found
Error relocating /bin/cat: __nanosleep_time64: symbol not found
Error relocating /bin/cat: __gettimeofday_time64: symbol not found
Error relocating /bin/cat: __[    9.027665] procd: - early -
settimeofday_tim[    9.032679] procd: - watchdog -
e64: symbol not found
Error relocating /bin/cat: __localtime64: symbol not found
Error relocating /bin/cat: __adjtimex_time64: symbol not found
Error relocating /bin/cat: __lstat_time64: symbol not found
Error relocating /bin/cat: __ctime64: symbol not found
Error relocating /bin/cat: __setitimer_time64: symbol not found
Error relocating /bin/grep: __gmtime64_r: symbol not found
Error relocating /bin/grep: __mktime64: symbol not found
Error relocating /bin/grep: __fstat_time64: symbol not found
Error relocating /bin/grep: __clock_settime64: symbol not found
Error relocating /bin/grep: __clock_gettime64: symbol not found
Error relocating /bin/grep: __utimes_time64: symbol not found
Error relocating /bin/grep: __futimens_time64: symbol not found
Error relocating /bin/grep: __utimensat_time64: symbol not found
Error relocating /bin/grep: __localtime64_r: symbol not found
Error relocating /bin/grep: __wait3_time64: symbol not found
Error relocating /bin/grep: __stat_time64: symbol not found
Error relocating /bin/grep: __time64: symbol not found
Error relocating /bin/grep: __nanosleep_time64: symbol not found
Error relocating /bin/grep: __gettimeofday_time64: symbol not found
Error relocating /bin/grep: __settimeofday_time64: symbol not found
Error relocating /bin/grep: __localtime64: symbol not found
Error relocating /bin/grep: __adjtimex_time64: symbol not found
Error relocating /bin/grep: __lstat_time64: symbol not found
Error relocating /bin/grep: __ctime64: symbol not found
Error relocating /bin/grep: __setitimer_time64: symbol not found
[    9.628477] procd: - watchdog -
[    9.631939] procd: - ubus -
[    9.785657] urandom_read: 4 callbacks suppressed
[    9.785663] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.799462] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.806606] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.813851] procd: - init -
Please press Enter to activate this console.
Error relocating /usr/libexec/loPlease press Enter to activate this console.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

deppen12 commented:
i want to compile this with new version of openwrt. i want to remove multiwan coz of single wan/lan port . also no use of wifi schedule.
